src/ring_theory/valuation/valuation_subring.lean

```diff
@@ -715,6 +722,18 @@ set.subset_set_smul_iff
 
 end pointwise_actions
 
+section
+
+variables {L : Type*} [field L]
+
+/-- The pullback of a valuation subring `A` along a ring homomorphism `K →+* L`. -/
+def comap (A : valuation_subring L) (f : K →+* L) :
+  valuation_subring K :=
+{ mem_or_inv_mem' := λ k, by simp [valuation_subring.mem_or_inv_mem],
+  ..(A.to_subring.comap f) }
```

Comment:
You should add the analogs of all the lemmas that we have for sub(semi)rings too:

https://github.com/leanprover-community/mathlib/blob/d1be7814498ffdc9e70d9a428e286b54df465eee/src/ring_theory/subsemiring/basic.lean#L411-L419